audio_service/lib/audio_service.dart

```diff
@@ -941,44 +940,68 @@ class AudioService {
     return handler;
   }
 
+  static Object? _artFetchOperationId;
   static Future<void> _observeMediaItem() async {
     await for (var mediaItem in _handler.mediaItem) {
-      if (mediaItem == null) continue;
+      if (mediaItem == null) {
+        continue;
+      }
+      final operationId = Object();
+      _artFetchOperationId = operationId;
       final artUri = mediaItem.artUri;
       if (artUri == null || artUri.scheme == 'content') {
-        await _platform.setMediaItem(
+        _platform.setMediaItem(
```

Comment:
Why is this no longer awaited?

Reply:
Yes, this was intentionally no longer awaited, because I noticed that updates were queueing. I just didn't realise that I could just replace `await for` with `.listen`, thanks. It makes a lot of sense to me now, I always thought there were no difference between them.